File: ValidParentheses/brilliant.java
public class Solution {
public boolean isValid(String s) {
    StringBuffer bf=new StringBuffer();
    int i=0;
    char temp;
    boolean flag=true;
    while(i<s.length())
    {
        temp=s.charAt(i);
        if(bf.length()!=0&&(temp==(char)(bf.charAt(bf.length()-1)+1)||temp==(char)(bf.charAt(bf.length()-1)+2)))
                bf.deleteCharAt(bf.length()-1);
        else
            bf.append(temp);
        i++;
    }
    if(bf.length()!=0)
        flag=false;
    return flag;
}
}

File: ValidParentheses/brilliant.c
void f(char **s,bool *b) {
    char cur=0;
    if(!**s||!*b)return;
    else cur=**s;
    switch(cur) {
        case ')':
        case ']':
        case '}':
            *b=false;
            return;
    }
    while(*b&&*++*s) {
        if(')'==**s||']'==**s||'}'==**s) {
            if(**s-cur>2||**s-cur<0) *b=false;
            else cur=0;
            break;
        } else f(s,b);
    }
    if(cur)*b=false;
}
bool isValid(char* s) {
    bool b=true;
    do {
        f(&s,&b);
    }while(b&&*s++);
    return b;
}
